pydantic/main.py

```diff
@@ -938,17 +943,32 @@ def _setattr_handler(self, name: str, value: Any) -> Callable[[BaseModel, str, A
                     _object_setattr(self, name, value)
                     return None  # Can not return memoized handler with possibly freeform attr names
 
-            cls._check_frozen(name, value)
-
             attr = getattr(cls, name, None)
             # NOTE: We currently special case properties and `cached_property`, but we might need
             # to generalize this to all data/non-data descriptors at some point. For non-data descriptors
             # (such as `cached_property`), it isn't obvious though. `cached_property` caches the value
             # to the instance's `__dict__`, but other non-data descriptors might do things differently.
+            if isinstance(attr, cached_property):
```

Comment:
Can you apply the same thing that I requested in https://github.com/pydantic/pydantic/pull/11431#discussion_r1952456838 ?